stdlib/internal/SwiftSyntax/TokenKind.swift.gyb

```diff
@@ -0,0 +1,109 @@
+%{
+  # -*- mode: Swift -*-
+  from gyb_syntax_support import *
+  # Ignore the following admonition it applies to the resulting .swift file only
+}%
+//// Automatically Generated From TokenKind.swift.gyb.
+//// Do Not Edit Directly!
+//===----------------- TokenKind.swift - Token Kind Enum ------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2017 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+/// Enumerates the kinds of tokens in the Swift language.
+public enum TokenKind: Codable {
+  case unknown
+  case eof
+% for token in SYNTAX_TOKENS:
+%   kind = token.swift_kind()
+%   if not token.text:
+%     kind += '(String)'
+%   end
```

Comment:
(this should go in a comment above this if statement)

If the token doesn't have an exact text specified, then the case will have a single associated value of type String. This is for `identifier`, `spacedBinaryOperator`, `unspacedBinaryOperator`, `prefixOperator`, `postfixOperator`, `integerLiteral`, `floatingLiteral`, and  `stringLiteral`